### PR TITLE
Migrate to latest immunization core library

### PR DESCRIPTION
**Pull request by ndegwamartin**
- Refactor to use combined vaccine map in immunization library
- Add support for processing more than 2 combined vaccines

### CODE REVIEW

File: opensrp-child/src/main/java/org/smartregister/child/task/SaveOutOfAreaServiceTask.java [hunk 1]
```diff
@@ -183,24 +184,17 @@ public static void addVaccine(VaccineRepository vaccineRepository, Vaccine vacci
...
-                ftsVaccineName = VaccineRepo.Vaccine.measles2.display();
-            }
 
-            if (ftsVaccineName != null) {
-                ftsVaccineName = VaccineRepository.addHyphen(ftsVaccineName.toLowerCase());
-                Vaccine ftsVaccine = new Vaccine();
-                ftsVaccine.setBaseEntityId(vaccine.getBaseEntityId());
-                ftsVaccine.setName(ftsVaccineName);
-                vaccineRepository.updateFtsSearch(ftsVaccine);
+            List<String> ftsVaccineNames = Utils.getCombinedVaccine(name, ImmunizationLibrary.getInstance().COMBINED_VACCINES_MAP);
```
Comment (ekigamba) [inline opensrp-child/src/main/java/org/smartregister/child/task/SaveOutOfAreaServiceTask.java]: I am thinking that we have some good functionality here that does a lot in updating the status for combined vaccine alternatives. This code is repeated [here](https://github.com/OpenSRP/opensrp-client-child/blob/63b1f56b30c9fc2c24938863c8d32bd6bab92a90/opensrp-child/src/main/java/org/smartregister/child/util/Utils.java#L204-L214) Do you think we can reuse it?

Also, I can't seem to figure out how the primary vaccine passed here is updated in the FTS search table? Any pointers?

File: opensrp-child/src/main/java/org/smartregister/child/util/Utils.java [hunk 2]
```diff
@@ -217,21 +225,28 @@ public static void addVaccine(VaccineRepository vaccineRepository, Vaccine vacci
...
+
+            String[] comboVaccines = StringUtils.stripAll(value.split("/"));
+
+            for (int i = 0; i < comboVaccines.length; i++) {
+
+
+                if (comboVaccines[i].trim().equals(vaccine_name)) {
+                    List<String> combos = Lists.newArrayList(comboVaccines);
+                    combos.remove(vaccine_name);
+                    return combos;
```
Comment (ekigamba) [inline opensrp-child/src/main/java/org/smartregister/child/util/Utils.java]: If not for future proofing I'd say we do a lot here, but it's OK

Above on the for-loop, I think we can use the map for quick lookup

File: opensrp-child/src/main/java/org/smartregister/child/task/SaveOutOfAreaServiceTask.java (hunk 1 above)
Comment (ndegwamartin) [reply opensrp-child/src/main/java/org/smartregister/child/task/SaveOutOfAreaServiceTask.java]: Yeah good catch, we can refactor this. 

The primary one is updated after being given in the call to  _vaccineRepository.add(vaccine)_

File: opensrp-child/src/main/java/org/smartregister/child/util/Utils.java (hunk 2 above)
Comment (ndegwamartin) [reply opensrp-child/src/main/java/org/smartregister/child/util/Utils.java]: Thanks, refactored

**Pull request merged**